test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -0,0 +1,207 @@
+# trademark licensing policies.
+
+from tests.interfaces import *
+
+
+@tags('multisched')
+class TestSchedulerInterface(TestInterfaces):
+
+    """
+    Test suite to test different scheduler interfaces
+    """
+
+    def setUp(self):
+        # delete all schedulers except default
+        self.server.manager(MGR_CMD_DELETE,
+                            SCHED,
+                            id="@default")
+        self.server.manager(MGR_CMD_CREATE,
+                            SCHED,
+                            id="TestCommonSched")
+        TestInterfaces.setUp(self)
+
+    def test_create_successful_scheduler_object(self):
+        """
+        Creates a scheduler object succcessfully .
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE,
+                                SCHED,
+                                {'sched_cycle_length': 12000},
+                                id="testCreateSched")
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_list_schedulers(self):
+        """
+        List a scheduler object.
+        """
+        pass
+
+    def test_delete_schedulers(self):
+        """
+        Delete a scheduler object succcessfully .
+        """
+        try:
+            self.server.manager(MGR_CMD_DELETE,
+                                SCHED,
+                                id="TestCommonSched")
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_duplicate_scheduler_name(self):
+        """
+        Check for the scheduler object name.
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE,
+                                SCHED,
+                                id="TestCommonSched")
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15211) returned from server' in e.msg[1])
+
+    def test_permission_on_scheduler(self):
+        """
+        Check for the permission to create/delete/modify scheduler object.
+        """
+        # Check for create permission
+        try:
+            self.server.manager(MGR_CMD_CREATE,
+                                SCHED,
+                                id="testCreateSched",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_CREATE,
+                                SCHED,
+                                id="testCreateSched",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+        # Check for delete permission
+        self.server.manager(MGR_CMD_CREATE,
+                            SCHED,
+                            id="testDeleteSched")
```

Comment:
You don't need to create a scheduler to delete.  You create one in setup.  Just delete that one.